sync/src/relayer/block_transactions_process.rs

```diff
@@ -61,9 +65,10 @@ impl<'a, CS: ChainStore + 'static> BlockTransactionsProcess<'a, CS> {
                     self.relayer
                         .accept_block(self.nc.as_ref(), self.peer, block);
                 }
+                return Ok(());
             }
         }
 
-        Ok(())
+        Err(Error::Misbehavior(Misbehavior::UnexpectedBlockTransactions).into())
```

Comment:
1. A send CompactBlock to B
2. B request for the corresponding GetBlockTransactions from A
3. B restart and lose its records.
4. A response B BlockTransactions. But A ban B unexpectly.

Am I getting it right?

Reply:
Yes. 
Before there were two different caches, so it was ok.

Reply:
I am not sure about it...

Reply:
anyway fixed it already.